WebRoot/public/launchZCS.jsp

```diff
@@ -484,6 +484,10 @@ delete text;
 			</c:if>
 		};
 
+		// Set the hab-root-id
+		batchInfoResponse.Header.context.refresh.HabRoot = "${domainInfo.attrs.zimbraHierarchicalAddressBookRoot}";
```

Comment:
instead of adding it in refresh block, you can create a global variable from this file which can be accessed anywhere
or better would be to add it as part of params on line 491

Reply:
I was thinking you can pass custom parameter from here to ZmZimbraMail and then in ZmZimbraMail#startup method we can get that root id and call requestmgr.getHabRoot, but I think it will too much deviate from current solution, so better to go ahead with the current approach, till we encounter some bug